src/serialization/chrono.hpp

```diff
@@ -72,15 +72,29 @@ inline auto parse_duration(const config_attribute_value& val, const Duration& de
 	return Duration{val.to_long_long(def.count())};
 }
 
-template<typename Rep, typename Period>
-constexpr auto deconstruct_duration(const std::chrono::duration<Rep, Period>& span)
+template<typename... Ts, typename Rep, typename Period>
+constexpr auto deconstruct_duration(const std::tuple<Ts...>&, const std::chrono::duration<Rep, Period>& span)
```

Comment:
What's the point of the tuple argument? IMO calls would be more readable without it.

Reply:
Allows the compiler to deduce the parameter pack by passing in a pre-set tuple. I figured it was cleaner than having to specify the template parameters every time, but I'm opening to refining the design.

Reply:
If the implication is that shorter is cleaner then I disagree. I don't see how
`auto [days, hours, minutes, seconds] = chrono::deconstruct_duration(chrono::format::days_hours_mins_secs, time_up)`
looks cleaner than
`auto [days, hours, minutes, seconds] = chrono::deconstruct_duration<chrono::days, std::chrono::hours, std::chrono::minutes, std::chrono::seconds>(time_up)`.
For the former besides looking up `deconstruct_duration` I also need to look up whatever `chrono::format::days_hours_mins_secs` is.